From 2a8200bba8d6372fdb2ffb38240542795084f7d8 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Sat, 24 Jan 2015 21:37:28 +0100
Subject: [PATCH 199/506] disk/ahci: Fix device_map_range argument.

Argument is not used on x86, hence it's gone unnoticed.

Found by: Coverity scan.
---
 grub-core/disk/ahci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/disk/ahci.c b/grub-core/disk/ahci.c
index 89365cd..d6bdbdd 100644
--- a/grub-core/disk/ahci.c
+++ b/grub-core/disk/ahci.c
@@ -198,7 +198,7 @@ grub_ahci_pciinit (grub_pci_device_t dev,
 		    | GRUB_PCI_COMMAND_MEM_ENABLED);
 
   hba = grub_pci_device_map_range (dev, bar & GRUB_PCI_ADDR_MEM_MASK,
-				   sizeof (hba));
+				   sizeof (*hba));
   grub_dprintf ("ahci", "dev: %x:%x.%x\n", dev.bus, dev.device, dev.function);
 
   grub_dprintf ("ahci", "tfd[0]: %x\n",
-- 
2.4.3

